### PR TITLE
fix: support `no_proxy` mode with camera-streamer (bookworm)

### PR DESCRIPTION
**Pull request by t3chguy**

### CODE REVIEW

**Comment by mryel00** [comment]
@t3chguy Thanks for the commit, but please have a look at my comment from #221 there are all the infos you need, before I can consider to merge it.
This is basically a duplicate but I didn't get a response from the OP of that other PR.

**Comment by t3chguy** [comment]
@mryel00 how about this?

File: libs/camera-streamer.sh [hunk 1]
```diff
@@ -36,11 +36,24 @@ function run_ayucamstream() {
...
     # set http port
     #
     start_param=( --http-port="${pt}" )
 
+    if [[ -n "${noprx}" ]] && [[ "${noprx}" = "true" ]]; then
+        # See https://github.com/mainsail-crew/crowsnest/pull/221#issuecomment-1863555700 for why
+        # we cannot assume the binary has support for `--http-listen`.
+        if $ust_bin --help | grep -q "http-listen"; then
+            start_param+=( --http-listen "0.0.0.0" )
+            log_msg "INFO: Set to 'no_proxy' mode! Using 0.0.0.0 !"
```
Comment (mryel00) [inline libs/camera-streamer.sh:50]: Small typo. Seems like ustreamer.sh has the same typo.
There shouldn't be a space in front of the `!`

File: libs/camera-streamer.sh [hunk 2]
```diff
@@ -36,11 +36,24 @@ function run_ayucamstream() {
...
+        else
+            log_msg "WARN: This version of camera-streamer lacks support for `no_proxy`"
```
Comment (mryel00) [inline libs/camera-streamer.sh:52]: Wrong apostrophes. Please use `'` to be consistent with our other messages (see e.g. two lines above).
Also add `!` at the end of the warning message.

**Review by mryel00 — CHANGES_REQUESTED** [review]

**Comment by mryel00** [comment]
Seems fine. I will make a sanity test tomorrow to be sure everything is correct.

**Comment by t3chguy** [comment]
Actually, looks like one of my printers is running such a version of camera-streamer, and it listens on `0.0.0.0` - so the WARN log is probably not needed.

**Comment by mryel00** [comment]
Yes they listen on `0.0.0.0` on the old version and you cannot change that.
You are right about the message. As long as it's `true` that message doesn't make any sense. For `false` it could make sense, but I don't think it's necessary tbh. So the warning can be deleted.

**Comment by t3chguy** [comment]
@mryel00 Done.

File: libs/camera-streamer.sh [hunk 3]
```diff
@@ -36,11 +36,21 @@ function run_ayucamstream() {
...
     ## construct start parameter
     # set http port
     #
     start_param=( --http-port="${pt}" )
 
+    if [[ -n "${noprx}" ]] && [[ "${noprx}" = "true" ]]; then
+        # See https://github.com/mainsail-crew/crowsnest/pull/221#issuecomment-1863555700 for why
+        # we cannot assume the binary has support for `--http-listen`.
+        if $ust_bin --help | grep -q "http-listen"; then
+            start_param+=( --http-listen "0.0.0.0" )
```
Comment (mryel00) [inline libs/camera-streamer.sh:49]: Did you actually test this? At least it doesn't work for me like this. You are missing a `=` here.
`--http-listen="0.0.0.0"` would be correct. I would even write `--http-listen=0.0.0.0`

**Review by mryel00 — CHANGES_REQUESTED** [review]
I tested it and this should be the last thing you need to change.
Also don't use force push please. I will squash merge the commit in the end and delete the commit history if it's useless. It just makes it harder for me to pull it if you force push it.

File: libs/camera-streamer.sh (hunk 3 above)
Comment (t3chguy) [reply libs/camera-streamer.sh:49]: You're right, locally I had just added `--http-listen="0.0.0.0"`

image

**Review by mryel00 — APPROVED** [review]

**Comment by mryel00** [comment]
Thank you for the contribution and the fast response time ❤️

**Pull request merged**